librabbitmq/amqp_openssl.c

```diff
@@ -662,8 +663,22 @@ destroy_openssl(void)
      * safely. We do leak the amqp_openssl_lockarray. Which is only
```

Comment:
Still unclear as to why it was done this way; this change doesn't make the problem worse, so I'm inclined to leave this as is for now.